recipe-server/client/control/components/MultiPicker.js

```diff
@@ -0,0 +1,234 @@
+  }
+
+  onTextChange(event) {
+    const { value } = event.target;
+
+    this.setState({
+      filterText: value,
+    });
+  }
+
+  getDisplayedOptions() {
+    const {
+      options = [],
+      value,
+    } = this.props;
+
+    const selectedOptions = options.filter(option =>
+      value.indexOf(option.value) === -1);
+
+    let displayedOptions = [].concat(selectedOptions)
+      .map(option => option.value)
+      .map(val =>
+        (!val ? null : { ...this.convertValueToObj(val) }))
+      .filter(x => x);
+
+    const {
+      filterText,
+    } = this.state;
+
+    if (filterText) {
+      displayedOptions = displayedOptions.filter(option =>
+        JSON.stringify(option).indexOf(filterText) > -1);
+    }
+
+    return displayedOptions;
+  }
+
+  handleApplyOption(event) {
+    event.persist();
+
+    if (!this.availableRef) {
+      return;
+    }
+
+    const {
+      value,
+      onChange,
+    } = this.props;
+
+    let selectedFilters = MultiPicker.getActiveSelectOptions(this.availableRef);
+
+    if (!selectedFilters || selectedFilters.length === 0) {
+      selectedFilters = this.getDisplayedOptions();
+      selectedFilters = selectedFilters.map(option => option.value);
+    }
+
+    const newOptions = []
+      .concat(value || [])
+      .concat(selectedFilters);
+
+    // clear user input
+    this.availableRef.value = null;
+
+    onChange(newOptions.join(','));
+  }
+
+  handleRemoveOption(event) {
+    event.persist();
+    if (!this.selectedRef) {
+      return;
+    }
+
+    const {
+      value,
+      onChange,
+    } = this.props;
+
+    const selectedFilters = MultiPicker.getActiveSelectOptions(this.selectedRef);
+    let newOptions = []
+      .concat(value || [])
+      .filter(val => selectedFilters.indexOf(val) === -1);
+
+    if (!selectedFilters || selectedFilters.length === 0) {
+      newOptions = [];
+    }
+
+    // clear the user selection
+    this.selectedRef.value = null;
+
+    onChange(newOptions.join(','));
+  }
+
+  convertValueToObj(value) {
+    const {
+      options = [],
+    } = this.props;
+
+    return options.find(option => option.value === value);
+  }
+
+  /**
+   * Render
+   */
+  render() {
+    const {
+      unit,
+      plural,
+      value,
+    } = this.props;
+
+    let pickerValue = value || [];
+
+    if (value && typeof value === 'string') {
+      pickerValue = pickerValue.split(',');
+    }
+
+    pickerValue = pickerValue.map(this.convertValueToObj);
+
+    const displayedOptions = this.getDisplayedOptions();
+
+    const availableLabel = displayedOptions
+      && (displayedOptions.length === 0 || displayedOptions.length > 1)
+      ? plural : unit;
+
+    const selectedLabel = pickerValue
+      && (pickerValue.length === 0 || pickerValue.length > 1)
+      ? plural : unit;
+
+    return (
+      <div className="multipicker">
+
+        <div className="mp-frame mp-from">
+          {`Available ${availableLabel}`}
```

Comment:
iirc this renders as one React 'text node', versus `Available {availableLabel}` which is two - one for 'available' and then one for the expression.

From the user's perspective, nothing is different. From the dev's, it's almost identical.